python/hsfs/core/feature_descriptive_statistics.py

```diff
@@ -25,116 +25,97 @@ class FeatureDescriptiveStatistics:
     def __init__(
         self,
         feature_type: str,
-        min: float,
-        max: float,
-        sum: float,
         count: int,
-        mean: float,
-        stddev: float,
+        end_time: Union[int, datetime, date, str],
+        row_percentage: int,
+        # for any feature type
         completeness: Optional[float] = None,
         num_non_null_values: Optional[int] = None,
         num_null_values: Optional[int] = None,
+        approx_num_distinct_values: Optional[int] = None,
+        # for numerical features
+        min: Optional[float] = None,
+        max: Optional[float] = None,
+        sum: Optional[float] = None,
+        mean: Optional[float] = None,
+        stddev: Optional[float] = None,
+        percentiles: Optional[Mapping[str, float]] = None,
+        # with exact uniqueness
         distinctness: Optional[float] = None,
         entropy: Optional[float] = None,
         uniqueness: Optional[float] = None,
-        approx_num_distinct_values: Optional[int] = None,
         exact_num_distinct_values: Optional[int] = None,
-        percentiles: Optional[Mapping[str, float]] = None,
-        id: Optional[int] = None,
+        # for filtering
         start_time: Optional[Union[int, datetime, date, str]] = None,
-        end_time: Optional[Union[int, datetime, date, str]] = None,
-        row_percentage: Optional[int] = None,
+        id: Optional[int] = None,
     ):
         self._id = id
         self._feature_type = feature_type
+        self._count = count
+        self._end_time = end_time
+        self._row_percentage = row_percentage
+        self._completeness = completeness
+        self._num_non_null_values = num_non_null_values
+        self._num_null_values = num_null_values
+        self._approx_num_distinct_values = approx_num_distinct_values
         self._min = min
         self._max = max
         self._sum = sum
-        self._count = count
         self._mean = mean
         self._stddev = stddev
-        self._completeness = completeness
-        self._num_non_null_values = num_non_null_values
-        self._num_null_values = num_null_values
+        self._percentiles = percentiles
         self._distinctness = distinctness
         self._entropy = entropy
         self._uniqueness = uniqueness
-        self._approx_num_distinct_values = approx_num_distinct_values
         self._exact_num_distinct_values = exact_num_distinct_values
-        self._percentiles = percentiles
         self._start_time = start_time
-        self._end_time = end_time
-        self._row_percentage = row_percentage
 
     @classmethod
     def from_response_json(cls, json_dict):
         json_decamelized = humps.decamelize(json_dict)
```

Comment:
If we allow scheduled statistics for whole feature group as discussed in the design log it could be quite convenient to have such method.

Reply:
Do you mean checking if json_dict contains multiple instances and return a list of deserialized `FeatureDescriptiveStatistics`?
If we allow scheduled statistics for the whole feature group, would that be reusing the same FM config without feature_name, or would `enable_feature_monitoring` create multiple FM configs, one per feature?
In the first case, we would need to modify the result - statistics relationship everywhere. In the second case, we wouldn't need to modify the relationship or this class.

Reply:
Ja i think maybe it makes more sense to modify at the FM result level. The relationship between FM results and configs allows already to do it without modifying FeatureDescriptiveStats so I think it makes sense